### PR TITLE
Full Review Request

### PR DESCRIPTION
**Pull request by rmarow**

### CODE REVIEW

**Comment by MattF-NSIDC** [comment]
Since you've got all the setup around linting, type checking, etc. set up already, I think you may as well set up a GitHub actions job to run those tests. Here's an example of a job that runs tests using `npm`, but you could adapt this to instead run tests using `invoke`. https://github.com/qgreenland-workshop-2023-researcher/qgreenland-workshop-2023-researcher.github.io/blob/main/.github/workflows/test.yml

File: noaa_metrics/aggregate_logs.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    if start_month == end_month:
+        if dataset != "all":
+            summary_header = f"NOAA Downloads {dataset} {start_month}\n\n"
+        else:
+            summary_header = f"NOAA Downloads {start_month}\n\n"
+    else:
+        if dataset != "all":
+            summary_header = f"NOAA Downloads {dataset} {start_month} - {end_month}\n\n"
+        else:
+            summary_header = f"NOAA Downloads {start_month} - {end_month}\n\n"
```
Comment (MattF-NSIDC) [inline noaa_metrics/aggregate_logs.py:192]: If possible, I'd like to combine this conditional tree with the one above. The ifs and elses are all the same, so maybe we could be setting the `subject`, `filename`, and `summary_header` all in one place.

**Review by MattF-NSIDC — CHANGES_REQUESTED** [review]

File: noaa_metrics/aggregate_logs.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    unique_users = unique_users_df.iloc[0][0]
+    total_download_bytes = total_download_bytes_df.iloc[0][0]
+    total_files = total_files_df.iloc[0][0]
+    summary = {
+        "Files Transmitted During Summary Period": total_files,
+        "Volume in MB of files Transmitted During Summary Period": total_download_bytes,
+        "Users Connecting During Summary Period": unique_users,
+    }
+    summary_df1 = pd.DataFrame.from_dict(summary, orient="index")
+    summary_df = summary_df1.rename(columns={0: "Values"})
```
[the remark this answers is not in the document]
Comment (rmarow) [reply noaa_metrics/aggregate_logs.py:56]: @MattF-NSIDC i initially had it that way, but inplace is being deprecated so thats where this method came in.

File: noaa_metrics/aggregate_logs.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    msg = EmailMessage()
+    msg["From"] = "archive@nusnow.colorado.edu"
+    msg["To"] = mailto
+    msg["Subject"] = subject
+
+    with open(full_report) as fp:
+        metrics_data = fp.read()
+    msg.add_attachment(metrics_data, filename=filename)
+    with smtplib.SMTP("localhost") as s:
+        s.send_message(msg)
```
[the remark this answers is not in the document]
Comment (rmarow) [reply noaa_metrics/aggregate_logs.py:165]: done

File: noaa_metrics/aggregate_logs.py (hunk 1 above)
Comment (rmarow) [reply noaa_metrics/aggregate_logs.py:194]: agreed! Done!

File: noaa_metrics/aggregate_logs.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    return month
+
+
+def get_year(date):
+    year = date.year
+    return year
+
+
+def email_full_report(full_report, year, start_month, end_month, mailto: str, dataset):
+    if start_month == end_month and dataset == "all":
```
[the remark this answers is not in the document]
Comment (rmarow) [reply noaa_metrics/aggregate_logs.py:143]: Resolved this by moving it and doing the nested conditional with the summery_header in the aggregate function.

File: noaa_metrics/cli.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
+import datetime as dt
+
+import click
+
+from noaa_metrics.ingest_logs import ingest_logs
+from noaa_metrics.aggregate_logs import aggregate_logs
+
+
+class DateType(click.ParamType):
```
[the remark this answers is not in the document]
Comment (rmarow) [reply noaa_metrics/cli.py:9]: done

File: noaa_metrics/aggregate_logs.py [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+def create_dataframe(
+    JSON_OUTPUT_DIR, start_date: dt.date, end_date: dt.date
+) -> pd.DataFrame:
+    """Create dataframe from JSON files."""
+    dates = pd.date_range(start_date, end_date, freq="d").strftime("%Y-%m-%d").tolist()
+    json_output_dir = os.fspath(JSON_OUTPUT_DIR)
+    files = [
+        f"{json_output_dir}/noaa-metrics-{date}.json"
+        for date in dates
+        if os.path.exists(f"{json_output_dir}/noaa-metrics-{date}.json")
```
[the remark this answers is not in the document]
Comment (rmarow) [reply noaa_metrics/aggregate_logs.py]: `FileNotFoundError: Some expected paths don't exist: [PosixPath('/share/logs/noaa-web/ingest/noaa-metrics-2023-01-02.json')]` 
This is the error that will now appear :)

**Pull request closed without merging**